h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -239,6 +264,33 @@ public void resize(int len) {
   private transient double _ds[];   // Doubles, for inflating via doubles
   public transient byte[]   _ss;   // Bytes of appended strings, including trailing 0
   private transient int    _is[];   // _is[] index of strings - holds offsets into _ss[]. _is[i] == -1 means NA/sparse
+    
+  public long byteSize() {
+    long mem = 0;
+    if (_ms != null) {
+      mem += _ms.byteSize();
+    }
+    if (_missing != null) {
+      mem += _missing.size() / 8; // size() returns number of bits, convert to bytes, approx
```

Comment:
Perhaps this is a bit optimistic?
One `BitSet` is at least 4 bytes for `wordsInUse`
https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/BitSet.java#L95
plus 4 bytes for `sizeIsSticky`
https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/BitSet.java#L101
plus 24 bytes for headers overhead for the `long[] words` ?
https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/BitSet.java#L90
see https://stackoverflow.com/questions/26874631/how-much-memory-does-a-java-array-use
I hope I get this right?
Would be great to see if all these estimates correlate well with actual jvm memory usage. 
Thanks !